### PR TITLE
gpsd: update to v3.19

### PR DESCRIPTION
**Pull request by RussellSenior**
Signed-off-by: Russell Senior <russell@personaltelco.net>

Maintainer: Pushpal Sidhu <psidhu.devel@gmail.com> / @psidhu 
Compile tested: ath79, TP-Link TL-WDR3600 v1, reboot-10619-g1a2d4d7317 (9437012b9ee4dc550e42665b71902cf885169100 + an scons update patch sent to openwrt-devel)
Run tested: no, I only need to build it because olsrd is sucking it in

Description:

updated version to solve a build problem (works with scons 3.1.0)

### CODE REVIEW

File: utils/gpsd/patches/010-musl.patch [hunk 1]
```diff
@@ -1,40 +0,0 @@
---- a/driver_ais.c
```
Comment (neheb) [inline utils/gpsd/patches/010-musl.patch]: I don't think this patch should be removed. It fixes compilation when compiling with -Werror=implicit-function-declaration.

Comment (neheb) [reply utils/gpsd/patches/010-musl.patch]: Never mind. Just saw that's it not needed anymore.

**Comment by sch-m** [comment]
LGTM.
gpsd is broken since openwrt/openwrt@316c8b5503e54cdb3ee5bd3b5d81a3b7dd139294

**Comment by RussellSenior** [comment]
@psidhu what do you mean by "the changelist"? There are 1482 commits between 3.17 and 3.19. I don't know what they are.

**Comment by psidhu** [comment]
@RussellSenior A nod to these would be good: 
* https://lists.nongnu.org/archive/html/gpsd-dev/2018-10/msg00005.html
* https://lists.nongnu.org/archive/html/gpsd-announce/2019-07/msg00000.html

**Comment by neheb** [comment]
URL should be updated to https://gpsd.gitlab.io/gpsd

**Comment by SvenRoederer** [comment]
I had recently pushed a similar commit to my repo (https://github.com/SvenRoederer/openwrt-packages/commit/46c83da92e06e24cb9f08daaacd6cd360fdcc436) , but was not able to PR it yet.

I added a note on dropping the "musl.patch":
Also dropping 010-musl.patch, as it seems to be fixed upstream with commit
c5d0f43cbdd69e2ed89b7e304c74208751be19c9 .

> Make sure gpsd_config.h is before all includes.
>
> This is for finicky OS like *BSD that need defines before
> includes like string.h.

**Comment by RussellSenior** [comment]
Anything blocking merging this?

**Comment by diizzyy** [comment]
Yes, no runtime test

**Comment by SvenRoederer** [comment]
The last update to v3.17 was comitted by @p-wassi . Probably he can do a runtime-test?

I've no GPS-hardware available to do it myself

**Comment by RussellSenior** [comment]
You can't even build it right now, without the update. How is building worse than not building, even if it doesn't run correctly, which seems unlikely given the minimal openwrt patching involved, isn't it better to provide something that builds so that we can get reports from actual users complaining that maybe it doesn't work right? Regrettably, I'm not in a position to test it. Not building is blocking other packages from building (e.g. olsrd) even if you haven't selected things that actually depend on it.

**Comment by neheb** [comment]
I think this should be updated. This is the master branch after all.

**Comment by diizzyy** [comment]
Let's wait 2 weeks for maintainer response before merging

**Comment by RussellSenior** [comment]
@psidhu do we need to wait 2 weeks for this?

**Comment by RussellSenior** [comment]
@diizzyy do you mean two weeks from now or two weeks from when I sent the pull request, since it's already been 9 days.

**Comment by RussellSenior** [comment]
Tested-by: @RussellSenior on ath79, using gpsmon built from the same package with a model BU-353 GPS device (via kmod-usb-serial-pl2303).

**Comment by psidhu** [comment]
If it's been tested, I have no issues merging this in.

**Comment by RussellSenior** [comment]
@diizzyy okay?

**Pull request merged**